src/MarketPlace.sol

```diff
@@ -34,8 +34,11 @@ contract MarketPlace {
         Notional // 8
     }
 
-    /// @notice markets are defined by a tuple that points to a fixed length array of principal token addresses.
-    mapping(address => mapping(uint256 => address[9])) public markets;
+    /// @notice markets are defined by a maturity and underlying tuple that points to an array of principal token addresses.
+    mapping(address => mapping(uint256 => address[])) public markets;
```

Comment:
we may want to create a struct and map both to an array of that object?  

Should save on gas considerably.

struct market {
address principal,
address adapter
}
mapping(address => mapping(uint256 => market[])) public markets;

Reply:
That makes sense, and has the added benefit of simplifying a bit of the code as well. I'll create a ticket to get to this in a separate PR for now